trade_plotter.py

```diff
@@ -0,0 +1,149 @@
+""" Trade Plotter: Plots a list of trades as html. """
+import datetime
+from typing import List
+
+import numpy as np
+import pandas as pd
+from timeutil import Time
+
+from backtesting_graphs._plotting import plot
+from backtesting_graphs._stats import compute_stats
+from backtesting_graphs.trade_model import Trade
+
+# TODO: These functions are already written in Cargo and should be imported here
+def make_dt(timestamp: int) -> datetime.datetime:
+    return datetime.datetime.fromtimestamp(int(timestamp), tz=datetime.timezone.utc)
+
+def get_candle_df(asset: str, start: Time, end: Time) -> pd.DataFrame:
+    """ Returns a data frame with candles for the given asset between the given
+        start and end times.
+
+        Args:
+            asset: Symbol of token to get candles for
+            start: Start time to get data from
+            end: End time to get data until, inclusive
+
+        Returns: Pandas dataframe indexed by time with candle columns """
+    # TODO (CASA-406): Pull data from Postgres
+    try:
+        df: pd.DataFrame = pd.read_csv(
+            f"data/{asset}-candles.csv", sep=",", index_col=0,
+            converters={"Timestamp": make_dt}
+        )
+    except FileNotFoundError as e:
+        print("Trying to get candles for %s failed because CSV file %s "
+                     "wasn't found.", f"{asset}-candles.csv", asset)
+        raise e
+
+    df = df.loc[start.dt:end.dt]
+    return df
+
+def graph_trades(trades: List[Trade], asset: str, start_time: Time,
+                 end_time: Time, init_cash: int, init_tokens: int) -> str:
+    """ Plots a list of trades for a given asset using the backtesting.py library
+        and returns the contents of the html file as a string.
+
+        Args:
+            trades: List of trades to plot
+            asset: Name of asset being plotted
+            start_time: Start time of candles
+            end_time: End time of candles
+            init_cash: Initial cash in portfolio prior to trading
+            init_tokens: Initial tokens in portfolio prior to trading """
+    candles_df = get_candle_df("eth", start_time, end_time)
+
+    num_candles_indices = int((end_time - start_time).total_seconds() // 60) + 1
+    calc_df = pd.DataFrame([], index=candles_df.index)
+    _gather_total_trade_deltas(
+        num_candles_indices, init_cash, init_tokens, calc_df, start_time, trades
+    )
+    _calculate_equity(calc_df, candles_df)
+
+    stats = compute_stats(
+        trades=trades, calc_df=calc_df, equity=calc_df.net_position_value,
+        ohlc_data=candles_df
+    )
+    filename = f"{asset}_trades_from_{start_time}_to_{end_time}"
+    # Set show_plot to False to not have the file outputted
+    return plot(results=stats, df=candles_df, filename=filename,
+                show_plot=True)
```

Comment:
I added this show_plot var just in case we wanted to see the plot directly sometimes. Should it default to False or True?